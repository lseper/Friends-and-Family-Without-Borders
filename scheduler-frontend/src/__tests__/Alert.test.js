import React from 'react';
import Alert from '../components/alert'
import "jest-enzyme"
import {shallow, configure, mount} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

configure({adapter: new Adapter()});
it('renders', () => {
    const tree = mount(
        <Alert
            message="Button"
            color="red-500"
        />
    )
    expect(tree).toMatchSnapshot()
});

it('displays the correct background color', () => {
    const wrapper = mount(
        <Alert
            message="Button"
            color="red-500"
        />
    )

    const firstDiv = wrapper.find('div').at(0)
    expect(firstDiv.hasClass('bg-red-500')).toBeTruthy()
});

it('displays the correct background color if no background color is passed in', () => {
    const wrapper = mount(
        <Alert
            message="Button"
        />
    )

    const firstDiv = wrapper.find('div').at(0)
    expect(firstDiv.hasClass("bg-")).toBeFalsy()
});

it('displays the correct text color', () => {
    const wrapper = mount(
        <Alert
            message="Button"
            color="red-500"
        />
    )

    const firstDiv = wrapper.find('div').at(0)
    expect(firstDiv.hasClass('text-white')).toBeTruthy()
});

it('displays the correct text when no message if passed in', () => {
    const wrapper = mount(
        <Alert
            color="red-500"
        />
    )

    const firstDiv = wrapper.find('span').at(1)
    expect(firstDiv.text()).toBe("")
});

it('displays the correct text', () => {
    const wrapper = mount(
        <Alert
            message= "Button"
            color="red-500"
        />
    )

    const firstDiv = wrapper.find('span').at(1)
    expect(firstDiv.text()).toBe("Button")
});

it('displays the exclamation point icon', () => {
    const wrapper = mount(
        <Alert
            message="Button"
            color="red-500"
        />
    )

    const numIcons = 1;
    const icon = wrapper.find('FontAwesomeIcon');
    expect(icon.length).toBe(numIcons);
});

it('check that button click updates use state to chose alert popup', () => {
    const setState = jest.fn();
    const useStateSpy = jest.spyOn(React, 'useState')
    useStateSpy.mockImplementation((init)=> [init, setState]);
    
    const wrapper = mount(
        <Alert
            message="Button"
            color="red-500"
        />
    )

    const xButton = wrapper.find('span').at(2);
    xButton.simulate("click");
    expect(setState).toHaveBeenCalledWith(false);
});

