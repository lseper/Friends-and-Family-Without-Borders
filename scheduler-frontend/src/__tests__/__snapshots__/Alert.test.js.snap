// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<Alert
  color="red-500"
  message="Button"
>
  <div
    className="text-white px-6 py-4 border border-white border-1 rounded relative bg-red-500"
  >
    <span
      className="text-white inline-block mr-5 align-middle"
    >
      <FontAwesomeIcon
        border={false}
        className="inline fa-lg mr-2 "
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              512,
              512,
              Array [],
              "f06a",
              "M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z",
            ],
            "iconName": "exclamation-circle",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-exclamation-circle fa-w-16 inline fa-lg mr-2 "
          data-icon="exclamation-circle"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </FontAwesomeIcon>
    </span>
    <span
      className="inline-block align-middle mr-8"
    >
      Button
    </span>
    <button
      className="absolute bg-transparent text-2xl font-semibold leading-none right-0 top-0 mt-4 mr-6 outline-none focus:outline-none"
      onClick={[Function]}
    >
      <span>
        Ã—
      </span>
    </button>
  </div>
</Alert>
`;
